codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -312,6 +315,14 @@ private void increaseInitialConnectionWindow(int deltaBytes) throws Http2Excepti
 
     final boolean consumeBytes(int streamId, int bytes) throws Http2Exception {
         Http2Stream stream = connection().stream(streamId);
+        // upgraded requests are ineligible for stream control
+        if (streamId == Http2CodecUtil.HTTP_UPGRADE_STREAM_ID) {
+            Boolean upgraded = stream.getProperty(upgradeKey);
+            if (upgraded != null && upgraded) {
+                return true;
```

Comment:
this method isn't documented very well (internal method I guess), but the methods it calls and the method which use it say `* @return {@code true} if a frame has been written as a result of this method call.`. This doesn't write a window update, so do we really want to return `true` here, and if so can we update the javadocs?

Reply:
I don't know.  @normanmaurer what do you think?

Reply:
the return status ultimately drives if we need to flush eventually or not to enable for "flush consolidation" ... so if there are no new frames generated I'm not sure why we would return `true`.

Reply:
That's compelling to me, I'm not sure I understand this well enough to draft a new comment though.  Do you mind suggesting a comment and I'll use that?

Reply:
My assumption is that we want to `return false;` here. If this is the case then we shouldn't need any additional comments (no WINDOW_UPDATE is written so returning `false` doesn't change existing behavior). If this is not the case then I don't understand why we need/want to return `true` so I don't yet have a suggestion.